finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -107,6 +107,7 @@ object Http extends Client[Request, Response] with HttpRichClient
     val stack: Stack[ServiceFactory[Request, Response]] =
       StackClient.newStack
         .replace(TraceInitializerFilter.role, new HttpClientTraceInitializer[Request, Response])
+        .replace(StackClient.Role.prepFactory, DelayedRelease.module)
```

Comment:
Do you also need `prepConn`?

I was going to propose making these changes look more like ClientBuilder, e.g.:

```
    val clientStack = {
      val stack0 = stack
        .replace(StackClient.Role.prepConn, prepConn)
        .replace(StackClient.Role.prepFactory, (next: ServiceFactory[Req, Rep]) =>
        codec.prepareServiceFactory(next))
```

But I think I like your way better, because it means less reliance on `Codec`.